### DISCUSSION TITLE
Promoting AIR to newcomers

### DISCUSSION THREAD

**Discussion by GeneralVimes** [airsdk/Adobe-Runtime-Support#3606] [General]
I was just looking through how other cross-platform development tools present themselves. 

For example, this is how Flutter presents itself to the newbie developers via such a fancy webpage: https://flutter.dev/

And a lesson of Your First Flutter App:
https://codelabs.developers.google.com/codelabs/flutter-codelab-first

What do you think of doing something like that for AIR? Having newcomers is essential for the development tool sustainability

Also looks like AIR wikipedia page strongly needs update: https://en.m.wikipedia.org/wiki/Adobe_AIR (probably mentioning notable releases post-201

**Comment by marchbold** [comment]
This is what we started to do with the [airsdk.dev](https://airsdk.dev/) domain. Obviously needs a lot of work but this was the intent. Unfortunately we don't have enough time to put more work into it than we have already but we continue to update it whenever we do.

**Comment by shaucky** [comment]
I hold pessimistic attitudes towards newcomers… It took me some time to think about this question.
 
**EDITED**: There was so many emotional speeches, now i think they should be hidden.
 
As a programming language, ActionScript is like my first love. I am very grateful to remember what Andrew wrote:
> Personally I would love to continue to do all this for the next 20 years, I like the technology and think it would be great if we can get it to the point where more and more people start to use it.
 
no matter how things went on.

**Comment by Shaunmax** [comment]
> this is how Flutter presents itself to the newbie developer 

Flutter is mostly targeting cross platform mobile app developers.  I dont think AIR is the best option for creating mobile apps. 
If you really want to get inspired check out what [Defold](https://defold.com) has achieved. 

 @marchbold do you have any plans to support Defold ? They totally lack good native support. Just curious to know!

**Reply by GeneralVimes** [reply]
Well, this is what specifically worked with AIR for me: with a single codebase I release my AIR games to iOS, Android, Windows and Linux. 
https://airapport.itch.io/farm-and-mine/devlog/720731/farm-and-mine-goes-fully-cross-platform

It would be cool for other game developer to know this opportunity

**Reply by Shaunmax** [reply]
Your work is truly inspiring for me, especially the effort you took for HTML5 conversion, but what about the new comers ?  Sometimes that extra effort alone might backfire and freak them out!

**Comment by seekerluke** [comment]
I've thought a lot about why I use AIR for games, and a lot of it comes down to "I just like it more". Flash and AIR have always had a really simple and gorgeous API, and other frameworks have tried to clone it to varying degrees of success. AIR is also really quick and easy to deploy to its supported platforms. A lot of game engines seem to struggle with slow build times, especially on mobile.

I sometimes see people say that we should drop Actionscript in favour of something else, but I think that's a mistake. I think a lot of AIR developers are using AIR because Actionscript is familiar, and changing that could drive away what's left of AIR's community.

If AIR wanted a niche beyond game development, I think a lot of people in the HTML5 world are getting tired of massive frameworks and tech stacks. We're already seeing this with projects like Svelte and HTMX that are trying to stay simple or lightweight. AIR and Feathers could fit into this niche, it offers one familiar language and straightforward GUI library with the option to branch out to native apps and games if desired. No need to learn 4+ languages and multiple libraries for a modern website.
